src/openms/include/OpenMS/APPLICATIONS/SearchEngineBase.h

```diff
@@ -0,0 +1,104 @@
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow $
+// $Authors: Chris Bielow $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/APPLICATIONS/TOPPBase.h>
+
+
+namespace OpenMS
+{
+  
+  /**
+    @brief Base class for Search Engine Adapters
+
+    It is build on top of TOPPBase and provides convenience functions for regular tasks in SearchEngines.
+
+    This base class enforces a common parameter scheme upon each adapter.
+    E.g. '-database' and '-in'.
+    This might be extended/changed in the future.
+    
+  */
+  class OPENMS_DLLAPI SearchEngineBase : public TOPPBase
+  {
+  public:
+    /// No default constructor
+    SearchEngineBase() = delete;
+
+    /// No default copy constructor.
+    SearchEngineBase(const SearchEngineBase&) = delete;
+
+    /**
+      @brief Constructor
+
+      Must match TOPPBase' Ctor!
+
+      @param name Tool name.
+      @param description Short description of the tool (one line).
+      @param official If this is an official TOPP tool contained in the OpenMS/TOPP release.
+      If @em true the tool name is checked against the list of TOPP tools and a warning printed if missing.
+
+      @param citations Add one or more citations if they are associated specifically to this TOPP tool; they will be printed during --help
+    */
+    SearchEngineBase(const String& name, const String& description, bool official = true, const std::vector<Citation>& citations = {}, bool toolhandler_test = true);
+
+    /// Destructor
+    virtual ~SearchEngineBase();
+
+    /**
+      @brief Reads the '-in' argument from internal parameters (i.e. the mzML file) and checks if MS2 spectra are present and are centroided. If no MS2 or profile MS2 data is found, an exception is thrown.
+
+      @param ms_level The MS level to check for their type (centroided/profile)
+
+      @return mzML filename (might be a relative or absolute path)
+
+      @throws OpenMS::Exception::FileEmpty if no spectra are found
+      @throws OpenMS::Exception::IllegalArgument if spectra are not centroided
+
+    */
+    String getRawfileName(int ms_level = 2) const;
+
+
+    /**
+      @brief Reads the '-database' argument from internal parameters (or from @p db) and tries to find the db in search directories (if it cannot be found immediately). If not found, an exception is thrown.
+      
+      @param db [Optional] Instead of reading the '-database', you can provide a custom name here (might be required for special db formats, see OMSSA)
+      @return filename for DB (might be a relative or absolute path)
+ 
+      @throws OpenMS::Exception::FileNotFound if database name could not be resolved
+    */
+    String getDBFilename(String db = "") const;
```

Comment:
is = "" needed?

Reply:
yes. For OMSSA, since we check the existance of the `.psq` file, even if only the fasta filename was provided in `-database`.

Reply:
ahh sorry, I misread. You mean, one has to pass the database filename explicitly for every adapter? Yeah... certainly possible.
I went with a default, so avoid repetitive code. But I have no strong feelings here.